aws/resource_aws_servicequotas_service_quota.go

```diff
@@ -0,0 +1,239 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/servicequotas"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsServiceQuotasServiceQuota() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsServiceQuotasServiceQuotaCreate,
+		Read:   resourceAwsServiceQuotasServiceQuotaRead,
+		Update: resourceAwsServiceQuotasServiceQuotaUpdate,
+		Delete: schema.Noop,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"adjustable": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"default_value": {
+				Type:     schema.TypeFloat,
+				Computed: true,
+			},
+			"quota_code": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"quota_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"request_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"request_status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"service_code": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"service_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"value": {
+				Type:     schema.TypeFloat,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsServiceQuotasServiceQuotaCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).servicequotasconn
+
+	quotaCode := d.Get("quota_code").(string)
+	serviceCode := d.Get("service_code").(string)
+	value := d.Get("value").(float64)
+
+	d.SetId(fmt.Sprintf("%s/%s", serviceCode, quotaCode))
+
+	input := &servicequotas.GetServiceQuotaInput{
+		QuotaCode:   aws.String(quotaCode),
+		ServiceCode: aws.String(serviceCode),
+	}
+
+	output, err := conn.GetServiceQuota(input)
+
+	if err != nil {
+		return fmt.Errorf("error getting Service Quotas Service Quota (%s): %s", d.Id(), err)
+	}
+
+	if output == nil {
+		return fmt.Errorf("error getting Service Quotas Service Quota (%s): empty result", d.Id())
+	}
+
+	if value > aws.Float64Value(output.Quota.Value) {
```

Comment:
Providing a value less than the current quota value would result in a noop and exit successfully, correct? If so should there be an error message to the operator that the provided value is less than the currently set quota?

Reply:
Correct. Terraform should show automatically show perpetual difference in this case. 😄